lua/octo/commands.lua

```diff
@@ -365,6 +368,14 @@ function M.setup()
       add = function()
         local current_review = require("octo.reviews").get_current_review()
         if current_review and utils.in_diff_window() then
+          -- if we have a current_review but no id, we are in browse mode.
+          -- for now, we cannot create comments.
+          -- TODO: implement 'non-review' commits here, which adds a diff commit
+          -- but outside of a review.
+          if current_review.id == -1 then
+            vim.notify("Please start or resume a review first", vim.log.levels.ERROR)
+            return
+          end
```

Comment:
I seem to still get the `gh: Could not resolve to a node with the global id of '-1'` error when trying to do a comment in the `browser` mode. I would assume this should catch that?

Reply:
![Screenshot 2025-02-21 at 11 25 12](https://github.com/user-attachments/assets/726847e5-3795-4b05-bd9c-a5829f3dee99)

Reply:
when you do this, do you already have a pending review ?

Reply:
hmmm @phdah I'm having trouble reproducing this.
If you run `octo review discard` and then try again, do you run into this?

Reply:
@ldelossa I still get the issue. I can try debugging it on my end.

Reply:
Okay, so your update here only works on the `Octo comment add`, but not for the `add_review_comment()`, which is default mapped to `<leader>ca`, found [here](https://github.com/ldelossa/octo.nvim/blob/ldelossa/octo-review-browse/lua/octo/mappings.lua#L133-L135).

I think your id check should be moved to the underlying `Review:add_comment()` method found [here](https://github.com/ldelossa/octo.nvim/blob/ldelossa/octo-review-browse/lua/octo/reviews/init.lua#L347), such that it looks like:
```lua
function Review:add_comment(isSuggestion)
  if self.id == -1 then
    vim.notify("Please start or resume a review first", vim.log.levels.ERROR)
    return
  end
  ...
```
Then the logic is general for all comment adding.

Reply:
With that change, it works for using both the `action` and `keybinding` for me at least 😄

Reply:
Hmm, is this kinda a code smell? The fact we have two different ways to create comments? It appears that `Oct comment add` is context specific and can discover if you're in a review or not? Should we **always** go thru this top-level command and stop allowing calls to `add_review_comment()`? 

@wd60622 @GuillaumeLagrange  would love some feedback on that.

Reply:
I think handled the scenarios where this error still may occur, going to resolve for now.